### PR TITLE
Photo diary

### PR DESCRIPTION
**Pull request by andygnewman**
Consolidates images from latest 4 photo diary 'articles' into an item in position 15 of the Word stream.

Also removes all photo diary entries from the stream.

Behind a flag - streamWorldPhotoDiary

screen shot 2015-07-20 at 17 27 37

### CODE REVIEW

**Comment by next-team** [comment]
I've spotted a [difference](https://s3-eu-west-1.amazonaws.com/ft-next-qa/image_diffs/stream-page/2015-07-17/15:32-2294/failures/index.html) between your pull request and production — [see full output](https://s3-eu-west-1.amazonaws.com/ft-next-qa/image_diffs/stream-page/2015-07-17/15:32-2294/successes/index.html)

File: client/components/stream/_photo-diary.scss [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	@include nTypeAlpha(3);
+	color: oColorsGetPaletteColor('warm-4');
+	margin: 5px 0;
+	text-align: center;
+	white-space: nowrap;
+	overflow: hidden;
+	text-overflow: ellipsis;
+	a:hover > &,
+	a:focus > & {
+		color: $next-blue;
```
Comment (i-like-robots) [inline client/components/stream/_photo-diary.scss]: I found there are mixins for link styles, https://github.com/Financial-Times/next-sass-setup/blob/master/_links.scss

File: server/lib/photo-diary.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+module.exports = function(streamItems) {
+
+	var metaData, brand, image;
+
+	function photoDiaryEntries(element, index, array) {
```
Comment (i-like-robots) [inline server/lib/photo-diary.js]: `index` and `array` vars aren't used

File: server/lib/photo-diary.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+		brand = metaData.brand[0];
+		image = element.raw.item.images[0];
+
+		if (brand && brand.term.name === 'FT Photo Diary') {
+			return false;
+		}
+		return element;
+	}
+
+	var latestPhotoDiaries = streamItems.filter(photoDiaryEntries).splice(0,4);
```
Comment (i-like-robots) [inline server/lib/photo-diary.js]: Do you mean `.slice()`?

File: views/partials/photo-diary.html [hunk 4]
```diff
@@ -0,0 +1,22 @@
+<section class="photo-diary" role="section">
+	<h2 class="stream__title">FT Photo Diary</h2>
+	<div class="o-grid-row">
+		{{#each items}}
+			<div data-o-grid-colspan="6 S6 M3" data-trackable="stream-photo-diary">
+				<div class="photo-diary__topic">
+					<a href="/{{id}}" class="photo-diary__link" data-trackable="photo-diary-link">
```
Comment (commuterjoy) [inline views/partials/photo-diary.html]: Would be worth mentioning to someone about linking to super hi-res versions. We'd need a full-width article template?

Comment (commuterjoy) [inline views/partials/photo-diary.html]: someone = chris?

**Comment by next-team** [comment]
I've spotted a [difference](https://s3-eu-west-1.amazonaws.com/ft-next-qa/image_diffs/stream-page/2015-07-20/08:52-2297/failures/index.html) between your pull request and production — [see full output](https://s3-eu-west-1.amazonaws.com/ft-next-qa/image_diffs/stream-page/2015-07-20/08:52-2297/successes/index.html)

**Comment by next-team** [comment]
I've spotted a [difference](https://s3-eu-west-1.amazonaws.com/ft-next-qa/image_diffs/stream-page/2015-07-20/09:27-2297/failures/index.html) between your pull request and production — [see full output](https://s3-eu-west-1.amazonaws.com/ft-next-qa/image_diffs/stream-page/2015-07-20/09:27-2297/successes/index.html)

**Comment by next-team** [comment]
I've spotted a [difference](https://s3-eu-west-1.amazonaws.com/ft-next-qa/image_diffs/stream-page/2015-07-20/11:10-2300/failures/index.html) between your pull request and production — [see full output](https://s3-eu-west-1.amazonaws.com/ft-next-qa/image_diffs/stream-page/2015-07-20/11:10-2300/successes/index.html)

**Comment by next-team** [comment]
I've spotted a [difference](https://s3-eu-west-1.amazonaws.com/ft-next-qa/image_diffs/stream-page/2015-07-20/15:36-2305/failures/index.html) between your pull request and production — [see full output](https://s3-eu-west-1.amazonaws.com/ft-next-qa/image_diffs/stream-page/2015-07-20/15:36-2305/successes/index.html)

**Comment by andygnewman** [comment]
Have changed styling to put it on a dark background as most sites do with photo galleries.

screen shot 2015-07-20 at 16 27 35

File: client/components/stream/_photo-diary.scss [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+	white-space: nowrap;
+	overflow: hidden;
+	text-overflow: ellipsis;
+}
+
+.photo-diary__headline-text {
+	@include nTypeAlpha(4);
+	@include nLinksHeadline();
+	@include oGridRespondTo(S) {
+		@include nTypeAlpha(3);
```
Comment (i-like-robots) [inline client/components/stream/_photo-diary.scss]: `nTypeAlphaSize`

File: server/controllers/v1-taxonomy.js [hunk 6]
```diff
@@ -163,6 +164,9 @@ module.exports = function(req, res, next) {
 									hotTopics = hotTopics.length > 1 ? hotTopics : false;
 								}
 							}
+							if (res.locals.flags.streamWorldPhotoDiary && req.params.thingId === 'MQ==-U2VjdGlvbnM=') {
```
Comment (i-like-robots) [inline server/controllers/v1-taxonomy.js]: Comment here

File: server/lib/photo-diary.js [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+		if (brand && brand.term.name === 'FT Photo Diary') {
+			return false;
+		}
+		return element;
+	}
+
+	var latestPhotoDiaries = streamItems.filter(photoDiaryEntries).slice(0,4);
+
+	if (latestPhotoDiaries.length === 4) {
+		var photoDiariesElement = {};
```
Comment (i-like-robots) [inline server/lib/photo-diary.js]: This assignment is a bit redundant, line 54 could be written as:

```js
streamItems.splice(14, 0, {
  template: 'photo-diary',
  items: latestPhotoDiaries.map(...)
});
```

**Comment by next-team** [comment]
I've spotted a [difference](https://s3-eu-west-1.amazonaws.com/ft-next-qa/image_diffs/stream-page/2015-07-20/16:33-2308/failures/index.html) between your pull request and production — [see full output](https://s3-eu-west-1.amazonaws.com/ft-next-qa/image_diffs/stream-page/2015-07-20/16:33-2308/successes/index.html)

File: client/components/stream/_photo-diary.scss [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+	text-decoration: none;
+}
+
+.photo-diary__intro-image {
+	position: absolute;
+	top: 0;
+	left: 0;
+	border: 0;
+	width: 100%;
+	// height: 100%;
```
Comment (i-like-robots) [inline client/components/stream/_photo-diary.scss]: min-height?

**Comment by i-like-robots** [comment]
Squerge!

**Pull request closed without merging**